embryo/segment_2D_0.py

```diff
@@ -111,3 +114,23 @@ def plot_comparison(plot1, plot2, title1, title2):
 thresholds = threshold_multiotsu(frame, classes=3)
 regions = np.digitize(frame, bins=thresholds)
 plot_comparison(frame, regions, "Original", "Multi-Otsu thresholding")
+
+# %%
+# Plot image histogram
+fig, ax = plt.subplots(ncols=1)
+
+ax.hist(frame.ravel(), bins=25)
+ax.set_title('Histogram')
+
+# %%
+# Enhance contrast by equalizing histogram
+enhanced_image = equalize_hist(frame)
+plt.imshow(enhanced_image, cmap='gray')
+
+# %%
+# Apply gradient magnitude using Sobel filter
+gradient = sobel(enhanced_image)
+# Apply a marker-based watershed algorithm
+markers = morphology.label(enhanced_image < 0.4 * enhanced_image.max())
```

Comment:
What kind of markers are you selecting here? From looking at the `markers` image, it's not clear to me at all...